centaur/src/main/scala/centaur/api/CentaurCromwellClient.scala

```diff
@@ -40,7 +41,19 @@ object CentaurCromwellClient {
     .getAs[Map[String, List[String]]]("centaur.metadata-args")
 
   def submit(workflow: Workflow): IO[SubmittedWorkflow] = {
-    sendReceiveFutureCompletion(() => cromwellClient.submit(workflow.toWorkflowSubmission(refreshToken = CentaurConfig.optionalToken)))
+    sendReceiveFutureCompletion(() => {
+      val submitted = cromwellClient.submit(workflow.toWorkflowSubmission(refreshToken = CentaurConfig.optionalToken))
+      submitted.biSemiflatMap(
```

Comment:
😮

Reply:
I couldn't tell if it was "bad" to `someIo.map { orig => log.info("Am I in 'correctly' in an IO?"); orig }`, or if the log should be also wrapped in an `IO(log.info(...))` and flatMap-ed. I went with the latter, possibly belt-and-suspenders.